### PR TITLE
[native] Generate timestamp filter based on plan

### PR DESCRIPTION
**Pull request by zacw7**
## Description
Generate Velox timestamp filter based on the Presto query plan.

## Motivation and Context
This is needed to add timestamp range for Presto Native.
https://github.com/facebookincubator/velox/issues/5807

## Impact
N/A

## Test Plan
Unit tests will be covered on the Velox-side changes.

## Release Notes

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Comment by amitkdutta** [comment]
Thanks @zacw7. Any chance we can add an e2e tests that exercises this plan conversion?

**Comment by zacw7** [comment]
> Thanks @zacw7. Any chance we can add an e2e tests that exercises this plan conversion?

This feature is not complete as it’s still missing Velox reader side change. There are 3 changes needed:
1. [Velox] Add timestamp filter support (https://github.com/facebookincubator/velox/pull/5808)
2. [Presto] Make sure the timestamp filter can be generated from the Presto plan during conversion. (https://github.com/prestodb/presto/pull/20554)
3. [Velox] Make the timestamp column reader to adapt the filter. (WIP)

File: presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp [hunk 1]
```diff
@@ -263,6 +271,29 @@ std::unique_ptr<common::HugeintRange> hugeintRangeToFilter(
   return std::make_unique<common::HugeintRange>(low, high, nullAllowed);
 }
 
+std::unique_ptr<common::TimestampRange> timestampRangeToFilter(
+    const protocol::Range& range,
+    bool nullAllowed,
+    const VeloxExprConverter& exprConverter,
+    const TypePtr& type) {
+  bool lowUnbounded = range.low.valueBlock == nullptr;
```
Comment (xiaoxmeng) [inline presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp:279]:
```
s/bool lowUnbounded/const bool hasLowerBound = range.low.valueBlock != nullptr;/
auto lowerBound = hasLowerBound ? toTimestamp(range.low.valueBlock, exprConverter, type) : std::numeric_limits<Timestamp>::min();
```

File: presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp [hunk 2]
```diff
@@ -263,6 +271,29 @@ std::unique_ptr<common::HugeintRange> hugeintRangeToFilter(
...
+  auto low = lowUnbounded
+      ? std::numeric_limits<Timestamp>::min()
+      : toTimestamp(range.low.valueBlock, exprConverter, type);
+  if (!lowUnbounded && range.low.bound == protocol::Bound::ABOVE) {
+    ++low;
+  }
+
+  bool highUnbounded = range.high.valueBlock == nullptr;
```
Comment (xiaoxmeng) [inline presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp:287]: ditto: the same as lower bound handling?

File: presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp [hunk 3]
```diff
@@ -219,6 +219,14 @@ int128_t toInt128(
   return value.value<velox::TypeKind::HUGEINT>();
 }
 
+Timestamp toTimestamp(
+    const std::shared_ptr<protocol::Block>& block,
+    const VeloxExprConverter& exprConverter,
+    const TypePtr& type) {
+  auto value = exprConverter.getConstantValue(type, *block);
```
Comment (xiaoxmeng) [inline presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp:226]:
```
nit: const. auto variant = ...
```

**Review by xiaoxmeng — COMMENTED** [review]
@zacw7 thanks for the change % minors.

**Comment by zacw7** [comment]
> @zacw7 thanks for the change % minors.

Thanks for the review! The current implementation is mostly identical to the existing code in the same file. I think it's no harm to add `const` to the variants, but wondering if we should rename them that might break the consistency.

**Review by xiaoxmeng — APPROVED** [review]

**Pull request merged**